pmm-app/tests/QAN/pages/qanPage.js

```diff
@@ -0,0 +1,136 @@
+const { I } = inject();
+const assert = require('assert');
+
+module.exports = {
+  url: 'graph/d/pmm-qan/pmm-query-analytics?from=now-5m&to=now',
+  filterGroups: [
+    'Environment',
+    'Cluster',
+    'Replication Set',
+    'Database',
+    'Node Name',
+    'Service Name',
+    'User Name',
+    'Node Type',
+    'Service Type',
+  ],
+  fields: {
+    table: '//table//tr[2]',
+    nextPageNavigation: '//ul[@role=\'navigation\']//li[last()]',
+    iframe: '//div[@class=\'panel-content\']//iframe',
+    newQANPanelContent: '.panel-content',
+    disabledResetAll: '//button[@data-qa="qan-filters-reset-all" and @disabled ]',
+    newQANSpinnerLocator: '//i[@data-qa=\'loading-spinner\']',
+    showSelected: '//button[@data-qa=\'qan-filters-show-selected\']',
+    countOfItems: '//span[@data-qa=\'qan-total-items\']',
+    filterBy: '//input[@data-qa=\'filters-search-field\']',
+    filterCheckboxes: '.checkbox-container__checkmark',
+    newQANAddColumn: '//span[contains(text(), \'Add column\')]',
+    searchFieldForColumn: 'input.ant-select-search__field',
+    newQANMetricDropDown: '.ant-select-dropdown-menu-item',
+    groupBySelector: '.group-by-selector',
+    addColumnSelector: '.add-columns-selector',
+    removeColumnButton: '//div[text()=\'Remove column\']',
+    resultsPerPageValue: '.ant-select-selection-selected-value',
+    nextPage: '.ant-pagination-next',
+    previousPage: '.ant-pagination-prev',
+    ellipsisButton: '.ant-pagination-item-ellipsis',
+    tableRow: 'div.tr',
+    resultPerPageCombobox: '.ant-pagination-options',
+    addColumnNewQAN: '//span[contains(text(), \'Add column\')]',
```

Comment:
```suggestion
    addColumnNewQAN: '//span[contains(text(), "Add column")]',
```

Reply:
Actually, I won't comment each of them, but pls consider this approach and maybe apply it in case it works.